src/manifest.py

```diff
@@ -476,15 +480,42 @@ def update_manifests(self) -> t.List[str]:
         # insertion-order-preserving dictionary so we use that.
         changes: t.Dict[str, t.Any]
         changes = OrderedDict()
-        for path, datas in self._external_data.items():
-            self._update_manifest(path, datas, changes)
-        if changes:
-            try:
-                self._update_appdata()
-            except AppdataNotFound as err:
-                log.info("Not updating appdata: %s", err)
-            except AppdataError as err:
-                self._errors.append(err)
-                log.error(err)
+
+        # if is-critical-source is found, only update the manifest if at least one source with is-critical-source received an update.
+
+        # assume we need to update everything by default
+        critical_version_changed = True
```

Comment:
Setting `critical_version_changed` to `True` when there are no critical sources is a bit misleading. Maybe split it to two variables, e.g. `has_critical_source` and `critical_version_changed`? Or make it tri-state, with `None` value indication no critical sources.